### PR TITLE
CIP-0104? | Web-Wallet Bridge - Account public key

### PR DESCRIPTION
**Pull request by Scitz0**
A [CIP-30 extension](https://cips.cardano.org/cips/cip30/#cardanowalletnameenableextensionsextensionpromiseapi) that allows for a DApp (if allowed) to fetch the connected account public key. Utilizes yet to-be-merged CIP-30 namespace PR https://github.com/cardano-foundation/CIPs/pull/577.

---

([rendered proposal in branch](https://github.com/Scitz0/CIPs/blob/cip30-ext-accountpub/CIP-0104/README.md))

### CODE REVIEW

**Comment by rphair** [comment]
@Scitz0 looks proper so far but I'm waiting on review until https://github.com/cardano-foundation/CIPs/pull/587#pullrequestreview-1608562582 is addressed.

File: CIP-XXXX/README.md
Comment (Ryun1) [inline CIP-XXXX/README.md:32]: > this CIP should, but not enforced

Should they or not? IMO sharing public keys should not require permission.

File: CIP-XXXX/README.md
Comment (Ryun1) [inline CIP-XXXX/README.md:22]: Is there a need to support wallet's proving ownership of the shared account key?

Perhaps CIP-30's signed data just be used (if the dApp derives payment/stake keys)?

File: CIP-XXXX/README.md
Comment (Ryun1) [inline CIP-XXXX/README.md:30]: How should this proposal be versioned? see [specification from CIP-0001](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0001/README.md#structure).

**Review by Ryun1 — COMMENTED** [review]
Hey @Scitz0!

Thanks for this proposal, I am a fan of this 🤓

See comments!

File: CIP-XXXX/README.md
Comment (AndrewWestberg) [reply CIP-XXXX/README.md:32]: It's a bit subtle, but connecting a wallet tells the dApp about all the currently used utxos and addresses of a wallet. Sharing the account public keys tells the dApp about any currently used utxos and addresses along with PAST/FUTURE addresses.  It's basically asking for an updatable read-only copy of the user's wallet instead of just the current snapshot of a user's wallet.

File: CIP-XXXX/README.md
Comment (AndrewWestberg) [reply CIP-XXXX/README.md:22]: This would add extra friction. All we really need here is a popup asking for permission from the user.

File: CIP-XXXX/README.md
Comment (Scitz0) [reply CIP-XXXX/README.md:32]: I think wallets SHOULD request additional access for the reason Andrew mentioned. But I wasn't sure if CIP should mention it as enforced or just as a recommendation. This is why I expressed myself as I did. In addition to past/future addresses, it will also allow you to derive not just payment and stake keys but also upcoming DRep pub keys (CIP-95).

File: CIP-XXXX/README.md
Comment (Scitz0) [reply CIP-XXXX/README.md:22]: I think its enough to handle it through the normal enable() call, if this extension is requested, just show enable popup to either allow or disallow.

File: CIP-XXXX/README.md
Comment (Scitz0) [reply CIP-XXXX/README.md:30]: 🤔 ... any recommendation for wording on this? 
I'm open to any suggestion really.

**Comment by rphair** [comment]
@Scitz0 @AndrewWestberg we've had to change the delimiter character because the colon was causing YAML errors due to the mapping syntax.  If you see any problems with this please post here and/or in https://github.com/cardano-foundation/CIPs/pull/594.

**Comment by Scitz0** [comment]
> @Scitz0 @AndrewWestberg we've had to change the delimiter character because the colon was causing YAML errors due to the mapping syntax. If you see any problems with this please post here and/or in #594.

I saw that `|` character also caused some issues 😄  ... I'll just hold off on the final decision on this. 👍

**Comment by rphair** [comment]
thanks @Scitz0 - we're trying to settle the issue now, as summarised in https://github.com/cardano-foundation/CIPs/pull/594#issuecomment-1726624411

**Comment by rphair** [comment]
@Scitz0 the 2 pending wallet API proposals were confirmed as candidates at the CIP meeting today... congratulations & please rename your CIP content folder to `CIP-0104` 🤩

**Comment by Scitz0** [comment]
Should we wait on merge until https://github.com/cardano-foundation/CIPs/pull/601 is resolved? 
Or we can deal with that after?

There are backslashes here, and this is how it was done in CIP30, so I just mimicked that.

**Comment by rphair** [comment]
sure @Scitz0 I think it's reasonable to first get consensus about how to handle the backslash-escaping of the left brackets in general.  Now that we have that PR on CIP-0030 I think we'll get attention from more devs including those who have worked with the affected parsers directly.

**Comment by rphair** [comment]
@Scitz0 I've put this on next CIP meeting agenda (https://hackmd.io/@cip-editors/87) in the hope of moving this forward; I'll follow the lead of the more wallet-oriented CIP editor(s) & approve this as soon as it passes their technical review.

File: CIP-0104/README.md
Comment (Ryun1) [inline CIP-0104/README.md:32]:
```suggestion

This endpoint must request the users permission to share the public key.

```

**Comment by rphair** [comment]
Came up all hands in favour at yesterday's CIP Editors meeting.  Once https://github.com/cardano-foundation/CIPs/pull/588#discussion_r1585184860 is added I expect @Ryun1 will approve it & it would be nice for @Crypto2099 also to approve since he was happy about it at the meeting. :)

**Review by rphair — APPROVED** [review]
Came up all hands in favour at yesterday's CIP Editors meeting.  Once https://github.com/cardano-foundation/CIPs/pull/588#discussion_r1585184860 is added I expect @Ryun1 will approve it & it would be nice for @Crypto2099 also to approve since he was happy about it at the meeting.  So @Scitz0 we are looking at merging this very soon & it's still marked `Last Check` for the next meeting if not.

File: CIP-XXXX/README.md
Comment (Ryun1) [reply CIP-XXXX/README.md]: Ah no, what I meant was adding a separate endpoint where the wallet signs with this key to prove ownership.
But I dont think this is needed as existing signData could be used (albeit with extra steps for the verifier).

File: CIP-XXXX/README.md
Comment (Ryun1) [reply CIP-XXXX/README.md]: something like

"This CIP is not to be versioned using a traditional scheme, rather if any large technical changes are required then a new proposal must replace this one. Small changes can be made if they are completely backwards compatible with implementations, but this should be avoided."

File: CIP-0104/README.md
Comment (Ryun1) [reply CIP-0104/README.md:32]: > Is this comment best placed here in API specification or is it fine in rationale section?

sorry i should have been more clear
yes, this should probably be explicit in the Specification

SHOULD is probably fine, it cant be enforced

**Pull request merged**
